java/connector-node/risingwave-connector-service/src/main/java/com/risingwave/connector/source/core/DbzCdcEngineRunner.java

```diff
@@ -103,18 +114,16 @@ public void start() throws InterruptedException {
             return;
         }
 
-        // put a handshake message to notify the Source executor
-        var controlInfo =
-                GetEventStreamResponse.ControlInfo.newBuilder().setHandshakeOk(true).build();
-        engine.getOutputChannel()
-                .put(
-                        GetEventStreamResponse.newBuilder()
-                                .setSourceId(engine.getId())
-                                .setControl(controlInfo)
-                                .build());
         executor.execute(engine);
-        running.set(true);
-        LOG.info("engine#{} started", engine.getId());
+        // wait for the connector to be fully started
+        boolean startOk = startSignal.await(WAIT_TIMEOUT, TimeUnit.SECONDS);
```

Comment:
> Leverage the fact that debezium will emit a heartbeat message when the connector has subscribed to upstream

I noticed that the heartbeat interval for mysql is 60s and for pg is 5min. Does dbz guarantee to send a heartbeat message immediately on startup?

Reply:
> Does dbz guarantee to send a heartbeat message immediately on startup?

Although didn't confirm in code, based on my testing for pg and mysql, the answer is yes.